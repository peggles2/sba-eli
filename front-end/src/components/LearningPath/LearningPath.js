import React from "react";
import { Header, Divider, Grid, Button, Icon } from "semantic-ui-react";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";

import TopicSideBar from "../TopicSideBar/TopicSidebar";
import TopicContentView from "../TopicContentView/TopicContentView";
import LearningEvent from "../LearningEvent/LearningEvent";
import LearningPathBreadCrumb from "./LearningPathBreadcrumb";
import MetaTags from "../SEO/MetaTags";

import { getPathWithTopics } from "../../actions/learningPathActions";

import "./LearningPath.scss";

export class LearningPath extends React.Component {
  componentDidMount() {
    this.initialFunctions();
  }

  initialFunctions() {
    const id = this.props.match.params.id;
    this.props.dispatch(getPathWithTopics(id));
  }

  componentDidUpdate(prevProps) {
    if (this.props.isUserLoggedIn !== prevProps.isUserLoggedIn) {
      const id = this.props.match.params.id;
      this.props.dispatch(getPathWithTopics(id));
    }
    
    if(this.props.match.params.topicId !== prevProps.match.params.topicId && this.props.match.params.eventId !== prevProps.match.params.eventId) {
      this.initialFunctions();
    }
  }

  renderRightColumnContent() {
    const isLearningEvent =
      this.props.match.params.topicId && this.props.match.params.eventId;
    const { topicId, id } = this.props.match.params;

    if (isLearningEvent) {
      //send the topic Title for wrapper header purposes, break this into state?
      const topicTitle = this.props.topicsList.find(
        topic => topic && topic.id ? topic.id.toString() === topicId.toString() : false
      );
      return <LearningEvent topicTitle={topicTitle ? topicTitle.name : null} />;
    } else {
      return (
        <TopicContentView course_id={id} topicsList={this.props.topicsList} />
      );
    }
  }

  render() {
    const courseId = this.props.match.params.id;

    const sideBarProps = {
      course_id: courseId,
      topicsList: this.props.topicsList,
      module_id: this.props.match.params.topicId,
      event_id: this.props.match.params.eventId
    };

    const breadCrumbProps = this.props.match.params;

    return (
      <Grid centered>
        <MetaTags
          metaTitle={this.props.learningPath.name}
          metaDescription={this.props.learningPath.description}
          canonicalUrl={`https://sba.gov/learning_paths/${
            this.props.match.params.id
          }`}
        />
        <Grid.Row className={"path-breadcrumb-row"}>
          <Grid.Column mobile={15} tablet={15} computer={15}>
            <LearningPathBreadCrumb
              pathName={this.props.learningPath.name}
              {...breadCrumbProps}
            />
          </Grid.Column>
        </Grid.Row>
        <Grid.Row className={"path-header-row"}>
          <Grid.Column mobile={15} tablet={15} computer={15}>
            <Header as="h2" className={"path-header"}>
              {this.props.learningPath.name}
              <Button className={"path-header-share mobile hidden"}>
                Share &nbsp; <Icon name={"share"} />
              </Button>
            </Header>
          </Grid.Column>
        </Grid.Row>
        <Divider />
        <Grid.Row>
          <Grid.Column
            className={"mobile hidden"}
            tablet={5}
            computer={5}
            largeScreen={3}
            widescreen={3}
          >
            <TopicSideBar {...sideBarProps} />
          </Grid.Column>
          <Grid.Column
            mobile={15}
            tablet={11}
            computer={11}
            largeScreen={11}
            widescreen={11}
          >
            {this.renderRightColumnContent()}
          </Grid.Column>
        </Grid.Row>
      </Grid>
    );
  }
}

const mapStateToProps = store => {
  return {
    learningPath: store.learningPath.learningPath,
    topicsList: store.learningPath.topicsList,
    isUserLoggedIn: store.login.isUserLoggedIn
  };
};

export default withRouter(connect(mapStateToProps)(LearningPath));