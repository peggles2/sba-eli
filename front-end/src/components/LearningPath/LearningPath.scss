@import "../../assets/styles/base";

.path-header {
  color: $sba-navy !important;
}

.path-header-share {
  display: inline-block;
  float: right;
}

.path-header-row {
  padding-top: 0 !important;
  padding-bottom: 0 !important;
}

.path-breadcrumb-row {
  margin-top: 1em !important;
}
