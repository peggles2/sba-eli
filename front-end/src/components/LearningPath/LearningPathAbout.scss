.learning-path-about-icon {
  margin-right: 10px;
}

.learning-path-about-icon-unselected {
  color: gray;
}

.learning-path-about-icon-selected {
  color: green;
}