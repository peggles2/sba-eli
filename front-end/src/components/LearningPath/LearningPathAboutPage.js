import React from 'react'
import { Button, Container, Header, Icon, Image, Modal } from 'semantic-ui-react'

const LearningPathAboutPage = (props) => {
  let placeholderText = "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras a elit non massa blandit egestas ac nec quam. Phasellus volutpat libero commodo consectetur scelerisque. Pellentesque efficitur eu libero sit amet rutrum. Aenean magna mauris, blandit a imperdiet vel, pellentesque ac sem.";
  
  let page = {
    1: {
      title: "Welcome",
      image: "https://react.semantic-ui.com/images/avatar/large/rachel.png",
      content: <Container>
        <p>
          { placeholderText }
        </p>        
      </Container>,
      controls: <Button onClick={props.nextPage}>Next</Button>
    },
    2: {
      title: "About",
      image: "https://react.semantic-ui.com/images/avatar/large/rachel.png",
      content: <Container>
        <p>
          { placeholderText }
        </p>
      </Container>,
      controls: <Button onClick={props.nextPage}>Next</Button>
    },
    3: {
      title: "Save Your Progress",
      image: "https://react.semantic-ui.com/images/avatar/large/rachel.png",
      content: <Container>
        <p>
          { placeholderText }
        </p>
        <Container textAlign='center'>
          
        </Container>
      </Container>,
      controls: <div>
        <Button>Register</Button>
        <Button onClick={props.nextPage}>Skip For Now</Button> 
      </div>
    },
    4: {
      title: "Start Your Path",
      image: "https://react.semantic-ui.com/images/avatar/large/rachel.png",
      content:  <Container>
        <p>
          { placeholderText }
        </p>
        <Container textAlign='center'>
          
        </Container>
      </Container>,
      controls:
        <Button icon labelPosition='right'>
          Start First Learning Path
          <Icon name='right arrow' />
        </Button>    
    }
  }
  
  if(!page[props.page]) {
    return ( <div /> );
  }
  
  return(
    <Modal.Content image>
      <Image wrapped src={ page[props.page].image } />
      <Modal.Description>
        <Header as='h1'>{ page[props.page].title }</Header>
        { page[props.page].content }
        <Container textAlign="center" style={{marginTop: '50px'}}>
          { page[props.page].controls }
        </Container>
      </Modal.Description>
    </Modal.Content>
  )
}

export default LearningPathAboutPage;