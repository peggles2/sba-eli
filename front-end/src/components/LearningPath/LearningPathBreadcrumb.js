import React, { Component } from "react";
import { Breadcrumb } from "semantic-ui-react";
import { withRouter } from 'react-router-dom';
import { connect } from "react-redux";

class LearningPathBreadCrumb extends Component {
  render() {
    const { id } = this.props;
    return (
      <Breadcrumb>
        <Breadcrumb.Section onClick={() => this.props.history.push('/')}>Home</Breadcrumb.Section>
        <Breadcrumb.Divider icon="triangle right" />
        <Breadcrumb.Section onClick={() => this.props.history.push('/learning_paths/' + this.props.id)}>
          {this.props.pathName}
        </Breadcrumb.Section>
      </Breadcrumb>
    );
  }
}

export default connect((store) => {
  return {}
})(withRouter(LearningPathBreadCrumb));
