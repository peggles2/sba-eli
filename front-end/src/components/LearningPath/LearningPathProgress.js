import React, { Component } from "react";
import { Grid, Progress } from "semantic-ui-react";

class LearningPathProgress extends Component {

  progressPercent = (complete, total) => {
    return (complete / total) * 100;
  }

  eventsCompleted = ( complete, total ) => {
    return complete + ` out of ` + total + ` done!`;
  }

  render() {
    const complete = this.props.complete;
    const total = this.props.total;

    return (
      <Grid className="learning-path-progress">
        <Grid.Row className="learning-path-progress-bar-row">
          <Grid.Column>
            <Progress
              color="blue"
              size="tiny"
              percent={this.progressPercent(complete, total)}
              className="learning-path-progress-bar"
            />
          </Grid.Column>
        </Grid.Row>
        <Grid.Row className="learning-path-completed-row">
          <Grid.Column>
            <span>{ this.eventsCompleted(complete, total) }</span>
          </Grid.Column>
        </Grid.Row>
      </Grid>
    )
  }
}

export default LearningPathProgress;
