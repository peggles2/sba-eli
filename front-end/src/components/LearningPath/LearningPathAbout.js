import React from 'react'
import { Button, Grid, Icon, Modal } from 'semantic-ui-react'

import LearningPathAboutPage from './LearningPathAboutPage'

class LearningPathAbout extends React.Component {
  state = {
    page: 1,
    modalOpen: false
  }
  
  nextPage = () => this.setState({ page: this.state.page + 1 })
  
  setPage = (page) => this.setState({ page: page })
  
  handleOpen = () => this.setState({ modalOpen: true, page: 1 })

  handleClose = () => this.setState({ modalOpen: false })

  render() {
    return (
      <Modal
        trigger={<Button onClick={this.handleOpen}>About</Button>}
        open={this.state.modalOpen}
        onClose={this.handleClose}
      >
        <LearningPathAboutPage page={this.state.page} nextPage={this.nextPage} handleClose={this.handleClose} />
        <Modal.Actions>
          <Grid verticalAlign='middle'>
            <Grid.Row>
              <Grid.Column width={4} textAlign="left">
                { this.state.page < 4 ? <Button onClick={this.handleClose}>Skip</Button> : "" }
              </Grid.Column>
              
              <Grid.Column width={8} textAlign="center">
                <Icon name="circle" onClick={() => this.setPage(1)} style={{marginRight: '10px', color: this.state.page === 1 ? 'green' : 'gray'}} />
                <Icon name="circle" onClick={() => this.setPage(2)} style={{marginRight: '10px', color: this.state.page === 2 ? 'green' : 'gray'}} />
                <Icon name="circle" onClick={() => this.setPage(3)} style={{marginRight: '10px', color: this.state.page === 3 ? 'green' : 'gray'}} />
                <Icon name="circle" onClick={() => this.setPage(4)} style={{color: this.state.page === 4 ? 'green' : 'gray'}} />
              </Grid.Column>
            </Grid.Row>
          </Grid>
        </Modal.Actions>
      </Modal>
    )
  }
}

export default LearningPathAbout;