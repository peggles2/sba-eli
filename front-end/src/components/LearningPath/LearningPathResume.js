import React, { Component } from "react";
import { Button, Container, Divider, Grid, Header, Image, Item } from "semantic-ui-react";
import LearningPathProgress from "./LearningPathProgress";

export default class LearningPathResume extends Component {
  render () {
    const learningPathImage = `/Image_Placeholder.png`;
    const learningPathHeader = "Entreprenerurial Leadership";
    const learningPathDescription = "Maybe it means something more - something we can't yet understand."
    const complete = 2;
    const total = 5;

    return(
      <Container>
        <Grid>
          <Grid.Row>
            <Grid.Column>
              <Header as='h2'>Let's get back on your journey!</Header>
              <Divider />
            </Grid.Column>
          </Grid.Row>
          <Grid.Row className="learning-path-resume">
            <Grid.Column width={5} className="learning-path-resume-image">
              <Image size="large" src={learningPathImage} />
            </Grid.Column>
            <Grid.Column width={10} className="learning-path-resume-content">
              <Header as='h2'>{learningPathHeader}</Header>
              <p>{learningPathDescription}</p>
              <LearningPathProgress complete={complete} total={total} />
              <Divider />
              <Button primary style={{ float: "right" }}>Resume</Button>
            </Grid.Column>
          </Grid.Row>
        </Grid>
      </Container>
    )
  }
}
