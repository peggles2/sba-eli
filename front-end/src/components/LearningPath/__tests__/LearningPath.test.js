import React from 'react';
import LearningPath from '../LearningPath';
import LearningObjectivesList from '../../LearningObjective/LearningObjectivesList'
import { shallow } from 'enzyme';

describe('LearningPath', () => {
  it('should render correctly', () => {
    const match = { params: { id: 1 } }
    const wrapper = shallow(<LearningPath match={match}/>);
    
    expect(wrapper).toMatchSnapshot();
  });
  
  it('should render all parts', () => {
    const match = { params: { id: 1 } }
    const wrapper = shallow(<LearningPath match={match}/>);

    expect(wrapper.find('div').length).toEqual(1);
    expect(wrapper.find("h1").length).toEqual(1);
    expect(wrapper.find(LearningObjectivesList).exists()).toBe(true)
  });

});
