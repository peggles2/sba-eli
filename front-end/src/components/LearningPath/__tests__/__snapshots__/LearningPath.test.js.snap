// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LearningPath should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LearningPath
    match={
      Object {
        "params": Object {
          "id": 1,
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1 />,
        <h3>
          Learning Objectives
        </h3>,
        <LearningObjectivesList
          course_id={1}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Learning Objectives",
        },
        "ref": null,
        "rendered": "Learning Objectives",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "course_id": 1,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1 />,
          <h3>
            Learning Objectives
          </h3>,
          <LearningObjectivesList
            course_id={1}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Learning Objectives",
          },
          "ref": null,
          "rendered": "Learning Objectives",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "course_id": 1,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
